tools/hidl/hidlcomp.cpp

```diff
@@ -3479,8 +3479,22 @@ void EspMessageInfo::write_esp()
     indentOuts(-1,"}\n");
         
     indentOuts("if (flags & 0x01) {\n");
+    bool isExtComplexType = true; //a complex type with no children, no attribute and no parent
+    if (isExtSimpleType || getParentName() || hasNonAttributeChild() || (espm_type_==espm_response && getMetaInt("exceptions_inline", 0)))
```

Comment:
Not sure I understand the question. If getParentName() returns true, the parent cannot be empty because this is the child of the parent.